libsmtutil/CVC5Interface.cpp

```diff
@@ -0,0 +1,319 @@
+				return m_solver.mkVar(CVC5Sort(*sortSort->inner), n);
+			else
+				try
+				{
+					return m_solver.mkInteger(n);
+				}
+				catch (cvc5::CVC5ApiRecoverableException const& _e)
+				{
+					smtAssert(false, _e.what());
+				}
+				catch (cvc5::CVC5ApiException const& _e)
+				{
+					smtAssert(false, _e.what());
+				}
+		}
+
+		smtAssert(_expr.hasCorrectArity(), "");
+		if (n == "ite")
+			return arguments[0].iteTerm(arguments[1], arguments[2]);
+		else if (n == "not")
+			return arguments[0].notTerm();
+		else if (n == "and")
+			return arguments[0].andTerm(arguments[1]);
+		else if (n == "or")
+			return arguments[0].orTerm(arguments[1]);
+		else if (n == "=>")
+			return m_solver.mkTerm(cvc5::Kind::IMPLIES, {arguments[0], arguments[1]});
+		else if (n == "=")
+			return m_solver.mkTerm(cvc5::Kind::EQUAL, {arguments[0], arguments[1]});
+		else if (n == "<")
+			return m_solver.mkTerm(cvc5::Kind::LT, {arguments[0], arguments[1]});
+		else if (n == "<=")
+			return m_solver.mkTerm(cvc5::Kind::LEQ, {arguments[0], arguments[1]});
+		else if (n == ">")
+			return m_solver.mkTerm(cvc5::Kind::GT, {arguments[0], arguments[1]});
+		else if (n == ">=")
+			return m_solver.mkTerm(cvc5::Kind::GEQ, {arguments[0], arguments[1]});
+		else if (n == "+")
+			return m_solver.mkTerm(cvc5::Kind::ADD, {arguments[0], arguments[1]});
+		else if (n == "-")
+			return m_solver.mkTerm(cvc5::Kind::SUB, {arguments[0], arguments[1]});
+		else if (n == "*")
+			return m_solver.mkTerm(cvc5::Kind::MULT, {arguments[0], arguments[1]});
+		else if (n == "div")
+			return m_solver.mkTerm(cvc5::Kind::INTS_DIVISION, {arguments[0], arguments[1]});
+		else if (n == "mod")
+			return m_solver.mkTerm(cvc5::Kind::INTS_MODULUS, {arguments[0], arguments[1]});
+		else if (n == "bvnot")
+			return m_solver.mkTerm(cvc5::Kind::BITVECTOR_NOT, {arguments[0]});
+		else if (n == "bvand")
+			return m_solver.mkTerm(cvc5::Kind::BITVECTOR_AND, {arguments[0], arguments[1]});
+		else if (n == "bvor")
+			return m_solver.mkTerm(cvc5::Kind::BITVECTOR_OR, {arguments[0], arguments[1]});
+		else if (n == "bvxor")
+			return m_solver.mkTerm(cvc5::Kind::BITVECTOR_XOR, {arguments[0], arguments[1]});
+		else if (n == "bvshl")
+			return m_solver.mkTerm(cvc5::Kind::BITVECTOR_SHL, {arguments[0], arguments[1]});
+		else if (n == "bvlshr")
+			return m_solver.mkTerm(cvc5::Kind::BITVECTOR_LSHR, {arguments[0], arguments[1]});
+		else if (n == "bvashr")
+			return m_solver.mkTerm(cvc5::Kind::BITVECTOR_ASHR, {arguments[0], arguments[1]});
+		else if (n == "int2bv")
+		{
+			size_t size = std::stoul(_expr.arguments[1].name);
+			auto i2bvOp = m_solver.mkOp(cvc5::Kind::INT_TO_BITVECTOR, {static_cast<unsigned>(size)});
+			// CVC5 treats all BVs as unsigned, so we need to manually apply 2's complement if needed.
+			return m_solver.mkTerm(
+				cvc5::Kind::ITE,
+				{
+						m_solver.mkTerm(cvc5::Kind::GEQ, {arguments[0], m_solver.mkInteger(0)}),
```

Comment:
There seems to be double tabs here and inside the expression

Reply:
Should be fixed now.